parsimonious/expressions.py

```diff
@@ -272,7 +273,7 @@ def _uncached_match(self, text, pos, cache, error):
     def _regex_flags_from_bits(self, bits):
         """Return the textual equivalent of numerically encoded regex flags."""
         flags = 'tilmsux'
-        return ''.join(flags[i] if (1 << i) & bits else '' for i in xrange(6))
+        return ''.join(flags[i] if (1 << i) & bits else '' for i in range(6))
```

Comment:
On 21.09.2015 08:16, Matthias Urlichs wrote:
> On 21.09.2015 06:34, Erik Rose wrote:
>>
>> In parsimonious/expressions.py
>> <https://github.com/erikrose/parsimonious/pull/79#discussion_r39941304>:
>>
>> > @@ -272,7 +273,7 @@ def _uncached_match(self, text, pos, cache, error):
>> >      def _regex_flags_from_bits(self, bits):
>> >          """Return the textual equivalent of numerically encoded regex flags."""
>> >          flags = 'tilmsux'
>> > -        return ''.join(flags[i] if (1 << i) & bits else '' for i in xrange(6))
>> > +        return ''.join(flags[i] if (1 << i) & bits else '' for i in range(6))
>>
>> We should import |range| from |six.moves| so we don't materialize a
>> list in Python 2.
>>
>>
>
> Sure, but materializing a range(6) list is probably cheaper than a
> generator …
… also, looking at that code, this should be len(flags), not 6 …


-- 
-- Matthias Urlichs